File: hackerrank/project_euler/problem_3/largest_prime_factor.js
// https://www.hackerrank.com/contests/projecteuler/challenges/euler003/problem

const largestPrimeFactor = n => {
    let a = 2;

    while (a <= Math.floor(Math.sqrt(n))) {
        if (n % a === 0) {
            n /= a;
        } else {
            a++;
        }
    }
    return n;
};

File: hackerrank/project_euler/problem_3/largest_prime_factor.py
# https://www.hackerrank.com/contests/projecteuler/challenges/euler003/problem


def l_p_n(n):
    d = 2
    while d <= int(n**0.5):
        if n % d == 0:
            n /= d
        else:
            d += 1
    return int(n)
